src/main/java/com/derongan/minecraft/deeperworld/extensions/DeeperWorldExtensions.kt

```diff
@@ -0,0 +1,60 @@
+package com.derongan.minecraft.deeperworld.extensions
+
+import org.bukkit.Material
+import org.bukkit.entity.Entity
+import org.bukkit.entity.Item
+import org.bukkit.entity.LivingEntity
+import org.bukkit.entity.Player
+
+internal fun Entity.getVehicleRecursive(): Entity? {
+    var currentVehicle = vehicle
+
+    while (currentVehicle?.vehicle != null) {
+        currentVehicle = currentVehicle.vehicle
+    }
+
+    return currentVehicle
+}
+
+internal fun Entity.getPlayerPassengersRecursive(): List<Player>? {
+    return mutableListOf<Player>().let { playerList ->
+        if (passengers.size != 0) {
+            if (passengers.any { it is Player }) {
```

Comment:
I don't think there's a need to do this check, should be able to just filter by instance and add all.